### PR TITLE
[youku] Add new url regex

### LINKED ISSUE #14261: URL support for cloud.youku.com and player.youku.com
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.09.15*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2017.09.15**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [x] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: http://cloud.youku.com/services/sharev?vid=XMzAwNzQzMjcyOA==
- Single video: http://player.youku.com/embed/XMzAzMTU1NjAzMg==

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

- Visit is probably restricted in Mainland China.
- youku.com is a supported site in `extractor/youku.py`.

### PR DESCRIPTION
**Pull request by holymonson**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This fix #14261.

### CODE REVIEW

File: youtube_dl/extractor/youku.py [hunk 1]
```diff
@@ -22,9 +22,14 @@ class YoukuIE(InfoExtractor):
...
         (?:
             https?://(
-                (?:v|player)\.youku\.com/(?:v_show/id_|player\.php/sid/)|
-                video\.tudou\.com/v/)|
-            youku:)
+                v\.youku\.com/v_show/id_|
+                player\.youku\.com/player\.php/sid/|
+                player\.youku\.com/embed/|
+                cloud\.youku\.com/services/sharev\?vid=|
+                video\.tudou\.com/v/
```
Comment (dstftw) [inline youtube_dl/extractor/youku.py]: This no longer matches http://v.youku.com/player.php/sid/XNDgyMDQ2NTQw/v.swf and http://player.youku.com/v_show/id_XMTc1ODE5Njcy.html.

Comment (holymonson) [reply youtube_dl/extractor/youku.py]: In fact those are invalid, you can check the links and will find 404. The original regex is overmatching.

Comment (dstftw) [reply youtube_dl/extractor/youku.py]: I'm well aware of that.

Comment (holymonson) [reply youtube_dl/extractor/youku.py]: @dstftw So, what should I do to get this fixed?

Comment (dstftw) [reply youtube_dl/extractor/youku.py]: Make it match URLs from the first post obviously.

Comment (holymonson) [reply youtube_dl/extractor/youku.py]: All valid URLs are already listed in `_TESTS`, and they are all matched by the patched regex.
IMHO, URLs from the first post is invalid so they should not be testcases, and there is no reason to support URLs not in `_TESTS`.

**Pull request closed without merging**